backend/src/main/java/com/google/models/Anomaly.java

```diff
@@ -0,0 +1,150 @@
+      Map<Timestamp, MetricValue> dataPoints) {
+    this.timestampDate = timestampDate;
+    this.metricName = metricName;
+    this.dimensionName = dimensionName;
+    this.dataPoints = dataPoints;
+  }
+
+  public Timestamp getTimestamp() {
+    return timestampDate;
+  }
+
+  public String getMetricName() {
+    return metricName;
+  }
+
+  public String getDimensionName() {
+    return dimensionName;
+  }
+
+  public Map<Timestamp, MetricValue> getDataPoints() {
+    return dataPoints;
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (o == this) {
+      return true;
+    }
+
+    if (!(o instanceof Anomaly)) {
+      return false;
+    }
+
+    Anomaly target = (Anomaly) o;
+
+    return target.timestampDate.equals(timestampDate) && target.metricName.equals(metricName)
+        && target.dimensionName.equals(dimensionName) 
+        && dataPoints.entrySet().stream().allMatch(
+          e -> e.getValue().equals(target.dataPoints.get(e.getKey()))
+        );
+  }
+
+  @Override
+  public String toString() {
+    StringBuilder str = new StringBuilder("");
+    str.append("Timestamp: " + timestampDate + "\n");
+    str.append("Metric Name: " + metricName + "\n");
+    str.append("Dimension Name: " + dimensionName + "\n");
+    str.append("Datapoints: \n");
+    dataPoints.forEach((key, value) -> 
+      str.append(key + ": " + value + "\n")
+    );
+    return str.toString();
+  }
+
+  public static Anomaly getDummyAnomaly() {
+    return new Anomaly(Timestamp.getDummyTimestamp(1), DUMMY_METRIC_NAME, DUMMY_DIMENSION_NAME, 
+        DUMMY_DATA_POINTS);
+  }
+
+  public static Entity toEntity(Anomaly anomaly) {
+    Entity anomalyEntity = new Entity(ANOMALY_ENTITY_KIND);
+    anomalyEntity.setProperty(Timestamp.TIMESTAMP_PROPERTY, anomaly.timestampDate.toString());
+    anomalyEntity.setProperty(METRIC_NAME_PROPERTY, anomaly.metricName);
+    anomalyEntity.setProperty(DIMENSION_NAME_PROPERTY, anomaly.dimensionName);
+
+    EmbeddedEntity dataPointsEntity = new EmbeddedEntity();
+    anomaly.dataPoints.forEach((timestamp, metricValue) -> 
+        dataPointsEntity.setProperty(timestamp.toString(), metricValue.getValue()));
+    anomalyEntity.setProperty(DATA_POINTS_PROPERTY, dataPointsEntity);
+
+    return anomalyEntity;
+  }
+
+  /** TODO: Move Entity to Embedded Entity conversion somewhere else that all entity can share. */
+  public static EmbeddedEntity toEmbeddedEntity(Anomaly anomaly) {
+    Entity anomalyEntity = toEntity(anomaly);
+    EmbeddedEntity anomalyEmbeddedEntity = new EmbeddedEntity();
+    anomalyEmbeddedEntity.setKey(anomalyEntity.getKey());
+    anomalyEmbeddedEntity.setPropertiesFrom(anomalyEntity);
+    return anomalyEmbeddedEntity;
+  }
+
+  public static Anomaly toAnomaly(EmbeddedEntity anomalyEmbeddedEntity) {
+    EmbeddedEntity dataPointsEE = (EmbeddedEntity) anomalyEmbeddedEntity.getProperty(DATA_POINTS_PROPERTY);
+    Map<Timestamp, MetricValue> m = new HashMap<>();
```

Comment:
better variable name than m?

Reply:
yep, that's a bad name.
changed it to `dataPointsMap`